jokers/red_card.lua

```diff
@@ -0,0 +1,163 @@
+local colours = {
+    red = {
+        next = "blue",
+        loc_key = "j_folly_red_card",
+        pos = { x = 0, y = 2 },
+        message_colour = G.C.RED,
+    },
+    blue = {
+        next = "yellow",
+        loc_key = "j_folly_blue_card",
+        pos = { x = 1, y = 2 },
+        message_colour = G.C.BLUE,
+    },
+    yellow = {
+        next = "green",
+        loc_key = "j_folly_yellow_card",
+        pos = { x = 2, y = 2 },
+        message_colour = G.C.YELLOW,
+    },
+    green = {
+        next = "red",
+        loc_key = "j_folly_green_card",
+        pos = { x = 3, y = 2 },
+        message_colour = G.C.GREEN,
+    },
+}
+
+return {
+    key = "red_card",
+    name = "fj_red_card",
+    atlas = "folly_jokers",
+    pos = { x = 0, y = 2 },
+    config = {
+        extra = {
+            colour = "red",
+            triggers = 0,
+            mult = 3,
+            chips = 10,
+            money = 1,
+            probability = 0.2,
+        },
+    },
+    calc_dollar_bonus = function(self, card)
+        if card.ability.extra.colour == "yellow" then
+            return card.ability.extra.money * card.ability.extra.triggers
+        end
+    end,
+    add_to_deck = function(self, card, from_debuff)
+        if card.ability.extra.colour == "green" then
+            self.update_probabilities(card)
+        end
+    end,
+    remove_from_deck = function(self, card, from_debuff)
+        local old = card.ability.extra.colour
+        card.ability.extra.colour = "red"
+        card.ability.extra.triggers = card.ability.extra.triggers + 1
```

Comment:
this will upgrade the card when it's debuffed right?
you probably need to remove a trigger after updating probabilities or rework the update_probabilities function